src/mrinufft/operators/autodiff.py

```diff
@@ -127,15 +135,59 @@ def __init__(self, nufft_op, wrt_data=True, wrt_traj=False):
             # used for update also.
             self._samples_torch = torch.Tensor(self.nufft_op.samples)
             self._samples_torch.requires_grad = True
+        self.batch_size = batch_size
+        self.paired_batch_mode = batch_size is not None
```

Comment:
You do have `self.batch_size` always with you, do you really need another boolean?